### PR TITLE
Move part of `eos/utils/` into `eos/maths/` and `eos/models/` libraries

### PR DESCRIPTION
**Pull request by dvandyk**

### CODE REVIEW

File: eos/b-decays/lambdab-to-lambdac-l-nu.cc [hunk 1]
```diff
@@ -21,17 +21,17 @@
 #include <eos/form-factors/baryonic.hh>
 #include <eos/rare-b-decays/charm-loops.hh>
 #include <eos/b-decays/lambdab-to-lambdac-l-nu.hh>
-#include <eos/utils/complex.hh>
+#include <eos/maths/complex.hh>
```
Comment (mreboud) [inline eos/b-decays/lambdab-to-lambdac-l-nu.cc:24]: I know it is a bit annoying to do, but I would like to have the header in alphabetic order and classified by sub-directories...

File: eos/maths/polylog.hh [hunk 2]
```diff
@@ -3,7 +3,7 @@
 #ifndef EOS_GUARD_EOS_UTILS_POLYLOG_HH
 #define EOS_GUARD_EOS_UTILS_POLYLOG_HH 1
```
Comment (mreboud) [inline eos/maths/polylog.hh:4]: You changed the name of the flag for some files, but not all of them. Is there a reason for that?

File: eos/b-decays/lambdab-to-lambdac-l-nu.cc (hunk 1 above)
Comment (dvandyk) [reply eos/b-decays/lambdab-to-lambdac-l-nu.cc:23]: Done.

File: eos/maths/polylog.hh (hunk 2 above)
Comment (dvandyk) [reply eos/maths/polylog.hh]: I changed the ones I saw. Went through all files and fixed all include guards.

**Pull request merged**